action/library/src/test/java/io/knotx/fragments/action/library/cache/CacheActionTest.java

```diff
@@ -0,0 +1,352 @@
+import static io.knotx.fragments.action.library.cache.CacheTestUtils.CACHE_KEY;
+import static io.knotx.fragments.action.library.cache.CacheTestUtils.DO_ACTION_LOGS;
+import static io.knotx.fragments.action.library.cache.CacheTestUtils.INVOCATIONS_LOGS_KEY;
+import static io.knotx.fragments.action.library.cache.CacheTestUtils.LOGS_KEY;
+import static io.knotx.fragments.action.library.cache.CacheTestUtils.PAYLOAD_KEY;
+import static io.knotx.fragments.action.library.cache.CacheTestUtils.SOME_VALUE;
+import static io.knotx.fragments.action.library.cache.CacheTestUtils.doActionAppending;
+import static io.knotx.fragments.action.library.cache.CacheTestUtils.doActionFailed;
+import static io.knotx.fragments.action.library.cache.CacheTestUtils.doActionIdleWithLogs;
+import static io.knotx.fragments.action.library.cache.CacheTestUtils.doActionReturning;
+import static io.knotx.fragments.api.FragmentResult.ERROR_TRANSITION;
+import static io.knotx.junit5.assertions.KnotxAssertions.assertJsonEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.lenient;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import io.knotx.fragments.action.api.Action;
+import io.knotx.fragments.action.api.log.ActionLogLevel;
+import io.knotx.fragments.action.library.cache.operations.CacheLookup;
+import io.knotx.fragments.action.library.cache.operations.CacheStore;
+import io.knotx.fragments.api.Fragment;
+import io.knotx.fragments.api.FragmentContext;
+import io.knotx.fragments.api.FragmentResult;
+import io.knotx.server.api.context.ClientRequest;
+import io.reactivex.Maybe;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.junit5.VertxExtension;
+import io.vertx.junit5.VertxTestContext;
+import io.vertx.reactivex.core.MultiMap;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+@ExtendWith(VertxExtension.class)
+class CacheActionTest {
+
+  private static final String KEY_WITH_PLACEHOLDERS = "{header.x}-{param.y}-{payload.z}-{config.p}";
+  private static final String KEY_INTERPOLATED = "header-param-payload-config";
+
+  @Mock
+  private CacheLookup lookup;
+
+  @Mock
+  private CacheStore store;
+
+  @Test
+  @DisplayName("Expect cache key is interpolated")
+  void cacheKeyInterpolated(VertxTestContext testContext) {
+    lookupSucceeding();
+    Action tested = create(KEY_WITH_PLACEHOLDERS, doActionIdleWithLogs());
+
+    verifyActionResult(testContext, tested, richContext(),
```

Comment:
Very nice tests, this PR looks like great piece of work to share on Knot.x Community!